pkg/stacks/unpack.go

```diff
@@ -218,13 +238,97 @@ func (sp *StackPackage) IsNamespaced() bool {
 
 // SetApp sets the Stack's App metadata
 func (sp *StackPackage) SetApp(app v1alpha1.AppMetadataSpec) {
-	sp.Stack.Spec.AppMetadataSpec = app
+	app.DeepCopyInto(&sp.Stack.Spec.AppMetadataSpec)
+
 	sp.appSet = true
 }
 
-// SetBehavior sets the Stack's Behavior configuration
-func (sp *StackPackage) SetBehavior(behavior unstructured.Unstructured) {
-	sp.Behavior = behavior
+// SetStackConfig sets the Stack's Behavior using StackConfig
+func (sp *StackPackage) SetStackConfig(config unstructured.Unstructured) {
+	sp.StackConfig = config
+}
+
+func (sp *StackPackage) createBehaviorController(sd v1alpha1.StackDefinitionBehavior) appsv1.DeploymentSpec {
+	// TODO(displague) where does the name come from? behavior? cmd line arg?
+	// sp.Stack.GetName() is not available when this runs
+	controllerDeploymentName := "tc-" + "something"
+	controllerContainerName := controllerDeploymentName
+	controllerImageName := "bymafmaf/templating-controller:0.0.1"
+
+	// TODO(displague) Where do these files reside within the Stack?
+	// are sp.baseDir or sd.Source.Path useful here?
+	// are we always mounting a remote source image?
+	behaviorDirName := "/"
+	behaviorContentsVolumeName := "behaviors"
+	behaviorMountPoint := "/behaviors"
+	spec := appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": controllerDeploymentName}},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{"app": controllerDeploymentName},
+			},
+			Spec: corev1.PodSpec{
+				RestartPolicy: corev1.RestartPolicyNever,
```

Comment:
I'm getting an error saying `Never` is not supported. I believe it's because that's a `Deployment`, not a `Job`, hence it requires user to put a container image that's _restartable_